### ISSUE TITLE
load_model with custom regularizers and custom objective fails

### ISSUE THREAD

**Issue by FallenShard** [fchollet/keras#5279]
Greetings, 

I've been trying to implement the Fast Neural Style paper. I've made the transform net and appended VGG16 etc. In addition, I've also added custom regularizers to some layers in the network to calculate loss. Therefore, the objective function objective(y_true, y_pred) that I'm using is set to return 0 and the training is done with custom regularizers only, which provide the loss.

However, there's a problem when I try to load the model with load_model after I've serialized it with model.save(). The error that appears is related to DisconnectedInputError when Theano can't compute the gradients. I believe this comes from the fact that loss is zero when the model loads itself and therefore has no additional loss from regularizers. Passing in custom_objects doesn't help either, in fact, custom regularizer classes are completely ignored (their constructors are not called).

If I try not to save/load the model, training actually works. However, I can't train everything in one go, as it takes a long time. Similarly, saving weights only doesnt help as I still need the optimizer state to continue.

Is there a way to resolve this issue? I suppose I can provide a minimal python script, however it would be quite long.

I'm using the latest Keras version (1.2.1) and windows 10 with theano GPU support.

**Comment by leaprovenzano** [comment]
hi, 
 i'm not sure if this will help but I had a similar issue with custom metrics which was solved by passing a dict to `custom_objects` when loading the model... so in my case it was : 

```python
my_model  = load_model( 'mymodel.h5',  
                       custom_objects={ 'my_custom_metric1': my_custom_metric1,
                                        'my_custom_metric2': my_custom_metric2})  
```

apologies if i'm off base...

**Comment by mustgoplay** [comment]
I may be running into this issue soon.  Have you tried making the loss 1 since you're ignoring it anyways?

Is there a way to pass updates to extra weights in the layers that just simply act as storage?

**Comment by FallenShard** [comment]
I've used pickle to save the model, and while it's not recommended and fairly memory inefficient, it did get the job done.

**Comment by saltsuger** [comment]
@FallenShard I've met the same issue as you described, Could you tell me the detail of saving the model with **Pickle**, maybe give some code ,   Thank you!

**Comment by elaeon** [comment]
load_model have a 'custom_objects' option, if your custom objetive function is called 'my_custom_objetive' then: 
from my_app import my_custom_objetive
load_model(path, custom_objects={'my_custom_objetive': my_custom_objetive})
